src/libcore/fmt/num.rs

```diff
@@ -203,14 +211,15 @@ macro_rules! impl_Display {
                 // convert the negative num to positive by summing 1 to it's 2 complement
                 (!self.$conv_fn()).wrapping_add(1)
             };
-            let mut buf: [u8; 20] = unsafe { mem::uninitialized() };
+            let mut buf: [u8; 40] = unsafe { mem::uninitialized() };
```

Comment:
Correct value of [`ceil(log10 (2^128))`](http://www.wolframalpha.com/input/?i=ceil%28log10+%282%5E128%29%29) is 39, not 40.